### PR TITLE
docs(CONTRIBUTING): Be more explicit about contributions

### PR DESCRIPTION
**Pull request by parlarjb**
Based on discussions with a few people, we're trying to make more explicit our contribution process. Most of this will be obvious to people already contributing, as it's just writing down our current structure, but we've also added a few points to try to make lives easier on our designers.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,24 +10,68 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
-
-1. The component is re-usable. Multiple projects have a need for it.
-3. Requirements are well defined and documented.
-4. If code has been written, it meets our standards (or is close enough for an easy update).
-  - If code hasn’t been written, developers are available to write it (if not, make the suggestion via a Github Issue and we can pick up the work as time permits).
-5. It's specific to EncoreUI. It matches the EncoreUI style. It doesn't require a large new framework.
-6. There isn't already a design pattern for it. The functionality doesn't already exist.
-
-When in doubt about fit, ask in the EncoreUI chat room or open a new GitHub issue.
+If you haven't already, let the EncoreUI team know what your plans are. This is important so that time isn't spent by separate teams doing the same thing, and so that the team can get an initial round of feedback in before coding starts.
```
Comment (glynnis) [inline CONTRIBUTING.md]: > If you haven't already, let the EncoreUI team know what your plans are

Will people understand that this means creating an issue? Or did you intend to communicate something else here? I'm also wondering whether all folks will know who counts as "the EncoreUI team".

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/980/gallery.html)

File: CONTRIBUTING.md (hunk 1 above)
Comment (chriscantu) [inline CONTRIBUTING.md]: I agree with @glynnis.  The wording seems a little unclear.  If the intent is to drive to creating an issue, can we be more explicit?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # How to contribute
 
-We want to keep it as easy as possible to contribute changes Encore. There are a few guidelines that we need to follow so that we can have a chance of keeping on top of things.
+We want to keep it as easy as possible to contribute changes Encore, while making sure not to overburden the core EncoreUI team. There are a few guidelines that we need to follow so that we can have a chance of keeping on top of things.
```
Comment (CITguy) [inline CONTRIBUTING.md]: "changes **to** Encore"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -10,24 +10,68 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+
+The most complex PRs are usually "New Components". We've written up our criteria for a New Component PR. This is based on what we've found works best for our team, to minimize the time spent reviewing PRs.
+
+Currently, when adding a new component, our most [precious](https://dl.dropboxusercontent.com/u/2384988/onering.jpg) resource is the time of our designers. We want to ensure that the PR process is as streamlined for them as possible, and this is made explicit below.
+
+Note that while this process is listed as for "New Components", it also has implication for "Styles" PRs, "Component Revisions", and others. Let common sense be your guide!
+
+## New Components - PRs
+* **Prerequisites**: 
+    * [Issue created](https://github.com/rackerlabs/encore-ui/issues)
```
Comment (CITguy) [inline CONTRIBUTING.md]: "[Issue created] with the following items:"

Then nest the next four bullet points underneath this item.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -10,24 +10,68 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+    * Link to Invision designs if these exist
+    * Screenshot in Issue of final visual design
+    * For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design` label to request input
+    * When all technical discussion on the issue is complete, change the label to `Ready for Dev`. For visual components/changes, *only* a designer may make this change.
+    * New Components _must_ be created using our [Component Scaffolding](./guides/ui-setup.md#creating-a-new-component)
```
Comment (CITguy) [inline CONTRIBUTING.md]: I'd throw this to the top of the list for better readability with nesting introduced in previous comment.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -10,24 +10,68 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+* **Step 1**: Submitter includes screenshot of new component in PR description
+* **Step 2**: Comment with Design Sign-Off on final product - Design LGTM
+* **Step 3**: Checklist
+    * [Unit Tests](./guides/testing.md#component-tests-aka-unit-tests)
+    * [Page Objects updated](./guides/testing.md#convienience-page-objects)
+    * [Functional/Midway Tests updated](./guides/testing.md#midway-tests)
+    * [CSS Best Practices (this document needs an update)](./guides/css-styleguide.md)
+    * Component Documentation Updated (i.e. the `README.md` for the component)
+    * Encore Style Guide updated
```
Comment (CITguy) [inline CONTRIBUTING.md]: "if applicable"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -10,24 +10,68 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+* **Step 4**: Comment from submitter with their verification of Checklist
+* **Step 5**: Requested Feedback:
+    * Code Review topics:
+        * CSS Best Practices (ie: LESS variables)
+        * JavaScript / Angular Best Practices
+        * Maintainability of the Code Base
+        * JavaScript Documentation
+        * Test Coverage validation
+    * Keep an eye out for Labels added by reviewers (ex. "On Hold", "Needs Design", etc.)
```
Comment (CITguy) [inline CONTRIBUTING.md]: I'd throw this to the top of the list. It may get glanced over, being between the two sublists.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -10,24 +10,68 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+        * Major visual component not approved by Design
+* **Step 6**: 2 Dev LGTM's
+* **Step 7**: Squash Commit ([see here for more details](#finalizing-a-pull-request))
+* **Step 8**: Final Travis Build Verification
+* **Step 9**: MERGE IT!
 
 ### Get feedback early and often
 
-It's much better to ask for feedback on an unfinished idea than to recieve feedback on a finished one. If you're developing a new component, or updating an old one, post code as you write it. Don't wait until the Pull Request for a code design review.
+It's much better to ask for feedback on an unfinished idea than to receive feedback on a finished one. If you're developing a new component, or updating an old one, feel free to post code as you write it. But please add "DON'T MERGE" to the title of the PR, to let people know it's not quite ready.
```
Comment (CITguy) [inline CONTRIBUTING.md]: I'd change to "DO NOT MERGE" so it matches the label.

**Comment by parlarjb** [comment]
I need to add the markdown for the PR description checklist.

**Comment by parlarjb** [comment]
I'll hold off on merging this until https://github.com/rackerlabs/encore-ui/pull/997 is brought in. I'll then rebase against that, and update accordingly.

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/980/gallery.html)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # How to contribute
 
-We want to keep it as easy as possible to contribute changes Encore. There are a few guidelines that we need to follow so that we can have a chance of keeping on top of things.
+We want to keep it as easy as possible to contribute changes to Encore, while making sure not to overburden the core EncoreUI team. There are a few guidelines that we need to follow so that we can have a chance of keeping on top of things.
```
Comment (CITguy) [inline CONTRIBUTING.md]: Please update to `EncoreUI` for consistency.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -10,24 +10,67 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+        * When all technical discussion on the issue is complete, change the label to `Ready for Dev`. For visual components/changes, *only* a designer may make this change.
+* **Step 1**: Submitter includes screenshot of new component in PR description
+* **Step 2**: Comment with Design Sign-Off on final product - Design LGTM
+* **Step 3**: Checklist
+    * [Unit Tests](./guides/testing.md#component-tests-aka-unit-tests)
+    * [Page Objects updated](./guides/testing.md#convienience-page-objects)
+    * [Functional/Midway Tests updated](./guides/testing.md#midway-tests)
+    * [CSS Best Practices (this document needs an update)](./guides/css-styleguide.md)
+    * Component Documentation Updated (i.e. the `README.md` for the component)
+    * Encore Style Guide updated if applicable
```
Comment (CITguy) [inline CONTRIBUTING.md]: Please update to `EncoreUI` for consistency.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -10,24 +10,65 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+## PR Steps
+* **Prerequisites**: 
+    * New Components _must_ be created using our [Component Scaffolding](./guides/ui-setup.md#creating-a-new-component)
+    * [Issue created](https://github.com/rackerlabs/encore-ui/issues) with the following items:
+        * Link to Invision designs if these exist
+        * Screenshot in Issue of final visual design
+        * For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design` label to request input
+        * When all technical discussion on the issue is complete, change the label to `Ready for Dev`. For visual components/changes, *only* a designer may make this change.
+* **Step 1**: Submitter includes screenshot of new component in PR description (See ["Design Review of Pull Requests"](#design-review-of-pull-requests) below)
+* **Step 2**: Comment with Design Sign-Off on final product - Design LGTM
```
Comment (glynnis) [inline CONTRIBUTING.md]: Can you link to the "Design Review of Pull Requests" section of this same doc here? Once that's done, LGTM.

Comment (parlarjb) [inline CONTRIBUTING.md]: I've got that link one line above in `**Step 1**`, you'd like it on this line too? Or only on this line?

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -10,24 +10,65 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+
+The most complex PRs are usually "New Components". The "PR Steps" below describe our criteria for putting together a new component PR. For the other types, adjust the steps as necessary, let common sense be your guide!
+
+Currently, when adding a new component, our most precious resource is the time of our designers. We want to ensure that the PR process is as streamlined for them as possible, and this is made explicit below.
+
+## PR Steps
+* **Prerequisites**: 
+    * New Components _must_ be created using our [Component Scaffolding](./guides/ui-setup.md#creating-a-new-component)
+    * [Issue created](https://github.com/rackerlabs/encore-ui/issues) with the following items:
+        * Link to Invision designs if these exist
```
Comment (CITguy) [inline CONTRIBUTING.md]: `Link to Invision designs (if applicable)`

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -10,24 +10,65 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+        * Screenshot in Issue of final visual design
```
Comment (CITguy) [inline CONTRIBUTING.md]: This sublist is for an issue. It's repetitive saying "in Issue".  I'd go with `Screenshot(s) of final visual design`.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -10,24 +10,65 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+        * For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design` label to request input
+        * When all technical discussion on the issue is complete, change the label to `Ready for Dev`. For visual components/changes, *only* a designer may make this change.
```
Comment (CITguy) [inline CONTRIBUTING.md]: IMO, this sublist would make more sense in a section on how to prepare an Issue for development.

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -10,24 +10,65 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+    * Keep an eye out for Labels added by reviewers (ex. "On Hold", "Needs Design", etc.)
+    * Code Review topics:
+        * CSS Best Practices (ie: LESS variables)
+        * JavaScript / Angular Best Practices
+        * Maintainability of the Code Base
+        * JavaScript Documentation
+        * Test Coverage validation
+    * Create an issue for non-requested feedback & tag with type
+        * Types: visual design, architecture, feature request, etc.
+        * i.e. topics outside of the scope of the PR should be left for later. If a component already has design sign-off, the PR is not the place to question the design or ask for design changes
```
Comment (CITguy) [inline CONTRIBUTING.md]: Can this sublist be split to a section about providing PR feedback?

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -10,24 +10,65 @@ When submiting a bug report, please **include a set of steps to reproduce the is
...
+* **Step 2**: Comment with Design Sign-Off on final product - Design LGTM
+* **Step 3**: Checklist
+    * [Unit Tests](./guides/testing.md#component-tests-aka-unit-tests)
+    * [Page Objects updated](./guides/testing.md#convienience-page-objects)
+    * [Functional/Midway Tests updated](./guides/testing.md#midway-tests)
+    * [CSS Best Practices (this document needs an update)](./guides/css-styleguide.md)
+    * Component Documentation Updated (i.e. the `README.md` for the component)
+    * EncoreUI Style Guide updated if applicable
+* **Step 4**: Comment from submitter with their verification of Checklist
+* **Step 5**: Requested Feedback:
```
Comment (CITguy) [inline CONTRIBUTING.md]: Can this sublist be split to a section about providing PR feedback?

Comment (parlarjb) [inline CONTRIBUTING.md]: Done, please see "Providing PR Feedback"

File: CONTRIBUTING.md (hunk 13 above)
Comment (parlarjb) [inline CONTRIBUTING.md]: I've created a section "Preparing an Issue for development"

File: CONTRIBUTING.md (hunk 10 above)
Comment (glynnis) [inline CONTRIBUTING.md]: LOL
http://i.giphy.com/vwI4mYEHP8k0w.gif
:+1:  carry on

File: CONTRIBUTING.md (hunk 15 above)
Comment (CITguy) [inline CONTRIBUTING.md]: :+1:

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -8,26 +8,79 @@ Please submit any bugs you encounter when using EncoreUI to our [Github Issues T
...
+There are certain things we look for in PR Feedback, and certain items that can usually "wait until later".
+
+* Code Review topics:
+    * CSS Best Practices (ie: LESS variables)
+    * JavaScript / Angular Best Practices
+    * Maintainability of the Code Base
+    * JavaScript Documentation
+    * Test Coverage validation
+
+Non-requested feedback (Create an issue for non-requested feedback & tag with type):
```
Comment (CITguy) [inline CONTRIBUTING.md]: I'd make this an H4

```
#### Non-Requested Feedback
Please create an issue for non-requested feedback and tag with type:

...
```

**Comment by CITguy** [comment]
lgtm, pending recent comment

**Comment by parlarjb** [comment]
Merging when done, there's a LGTM from @glynnis embedded into a line comment above.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -8,26 +8,85 @@ Please submit any bugs you encounter when using EncoreUI to our [Github Issues T
...
+
+### Providing PR Feedback
+There are certain things we look for in PR Feedback, and certain items that can usually "wait until later".
+
+* Code Review topics:
+    * CSS Best Practices (ie: LESS variables)
+    * JavaScript / Angular Best Practices
+    * Maintainability of the Code Base
+    * JavaScript Documentation
+    * Test Coverage validation
```
Comment (CITguy) [inline CONTRIBUTING.md]: *`validation` (in the last bullet point) should be capitalized for consistency*

This section should contrast with `Non-Requested Feedback` below, so it should be labeled `Requested Feedback` to be more explicit.  I'd convert this list to the following:

```
#### Requested Feedback
* CSS Best Practices (ie: LESS variables)
* JavaScript / Angular Best Practices
* Maintainability of the Code Base
* JavaScript Documentation
* Test Coverage Validation
```

**Comment by CITguy** [comment]
LGTM, pending latest comment. (I mean it this time. :smile: )

**Pull request merged**